emake.mk

```diff
@@ -0,0 +1,101 @@
+# Minimum configuration
+#   PACKAGE_BASENAME: the basename that prefixes all Lisp files in this package
+
+ifndef PACKAGE_BASENAME
+$(error PACKAGE_BASENAME is not set)
+endif
+
+# User options
+#   EMACS_ARGS: [extra arguments for each invocation of emacs]
+#   (below; see README)
+PACKAGE_FILE            ?= $(PACKAGE_BASENAME).el
+PACKAGE_LISP            ?= $(filter-out %-pkg.el %-autoloads.el, $(wildcard $(PACKAGE_BASENAME)*.el))
+PACKAGE_TESTS           ?= $(wildcard test/*.el)
+PACKAGE_ARCHIVES        ?= gnu
+PACKAGE_TEST_ARCHIVES   ?= gnu
+
+# Then, make it easy to invoke Emacs with EMake loaded.
+EMAKE = PACKAGE_FILE="$(PACKAGE_FILE)" \
+	PACKAGE_LISP="$(PACKAGE_LISP)" \
+	PACKAGE_TESTS="$(PACKAGE_TESTS)" \
+	PACKAGE_ARCHIVES="$(PACKAGE_ARCHIVES)" \
+	PACKAGE_TEST_DEPS="$(PACKAGE_TEST_DEPS)" \
+	PACKAGE_TEST_ARCHIVES="$(PACKAGE_TEST_ARCHIVES)" \
+	emacs -batch -l emake.el \
+	--eval "(setq enable-dir-local-variables nil)" \
+	$(EMACS_ARGS) \
+	--eval "(emake (pop argv))"
+
+# Set up our phony targets so Make doesn't think there are files by
+# these names.
+.PHONY: clean setup install compile test help
+
+help:                           ## show help
```

Comment:
this doesn't seem to work for me

```
make help
emake.mk                       show help
emake.mk                       clean all generated files
emake.mk                       clean all generated files including scripts
emake.mk                       Commands useful for Travis
...
```

Reply:
Yeah I noticed that… I need to dig into how the pipeline is actually behaving. It works when there's only one Makefile in the list, but it seems to break here.

Reply:
Fixed! Added the `-h` option to the grep call.

Reply:
thanks